regexec.c

```diff
@@ -328,11 +330,70 @@ S_regcppush(pTHX_ const regexp *rex, I32 parenfloor, U32 maxopenparen _pDEPTH)
         PTR2UV(rex),                                                        \
         PTR2UV(rex->offs),                                                  \
         (UV)(ix),                                                           \
-        (IV)(rex)->offs[ix].start,                                          \
-        (IV)(rex)->offs[ix].end,                                            \
+        (IV)(rex)->offs[ix].start_new,                                      \
+        (IV)(rex)->offs[ix].end_new,                                        \
         (UV)(rex)->lastparen                                                \
     ))
 
+#define CAPTURE_CLEAR(from_ix, to_ix, str)  STMT_START {                \
```

Comment:
Well you can review the code for sanity. Right now that is much better than nothing. Also it doesnt help much if only me and Dave M know this code. What is the point of having code review if everybody says "I am not competent to review this?". 

Also, frankly i want to restructure how we store the paren offsets. I dont think using a single array of structs actually fits our needs very well. If it were stored as simple arrays it would be much more efficient to clear the paren buffers, or push them into the stack. To clear we really only need to set the ends to -1. (This is the kind of review that would be helpful.) 

We should organize the structure the operations of copying the paren data to the save stack, and clearing the parens (in two different ways) is efficient.  It seems to me if the paren data were organized lineararly, with starts, end, start_new, end_new and start_tmp all being separate chunks, then we would be able to clear the end_new and the end arrays independently with memset() (as we use -1 for the value), and we would be able to copy chunks of them into the save stack with memcpy.